rust/basic_ethereum/README.md

```diff
@@ -0,0 +1,179 @@
+
+```bash
+git clone https://github.com/dfinity/examples
+cd examples/rust/basic_ethereum
+git submodule update --init --recursive
+```
+
+**If you are using MacOS, you'll need to install Homebrew and run `brew install llvm` to be able to compile the example.
+**
+
+### Acquire cycles to deploy
+
+Deploying to the Internet Computer
+requires [cycles](https://internetcomputer.org/docs/current/developer-docs/setup/cycles) (the equivalent of "gas" in
+other blockchains). You can get free cycles from
+the [cycles faucet](https://internetcomputer.org/docs/current/developer-docs/setup/cycles/cycles-faucet.md).
+
+### Deploy the smart contract to the Internet Computer
+
+```bash
+dfx deploy --network ic basic_ethereum --argument '(opt record {ethereum_network = opt variant {Sepolia}; ecdsa_key_name = opt variant {TestKey1}})'
+```
+
+#### What this does
+
+- `dfx deploy` tells the command line interface to `deploy` the smart contract
+- `--network=ic` tells the command line to deploy the smart contract to the mainnet ICP blockchain
+- `--argument (opt record {ethereum_network = opt variant {Sepolia}; ecdsa_key_name = opt variant {TestKey1}})`
+  initializes the smart contract with the provided arguments:
+    - `ethereum_network = opt variant {Sepolia}`: the canister uses
+      the [Ethereum Testnet Sepolia](https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-11155111.json)
+      network.
+    - `ecdsa_key_name = opt variant {TestKey1}`: the canister uses a test key for signing via threshold ECDSA that is
+      available on the ICP mainnet.
+      See [signing messages](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/encryption/signing-messages#signing-messages-1)
+      for more details.
+
+If successful, you should see an output that looks like this:
+
+```bash
+Deploying: basic_ethereum
+Building canisters...
+...
+Deployed canisters.
+URLs:
+Candid:
+    basic_ethereum: https://a4gq6-oaaaa-aaaab-qaa4q-cai.raw.icp0.io/?id=<YOUR-CANISTER-ID>
+```
+
+Your canister is live and ready to use! You can interact with it using either the command line or using the Candid UI,
+which is the link you see in the output above.
+
+In the output above, to see the Candid Web UI for your ethereum canister, you would use the
+URL `https://a4gq6-oaaaa-aaaab-qaa4q-cai.raw.icp0.io/?id=<YOUR-CANISTER-ID>`. You should see the methods specified in the Candid file `basic_ethereum.did`.
+
+## Step 2: Generating an Ethereum address
+
+An Ethereum address can be derived from an ECDSA public key. To derive a user's specific address, identified on the IC
+by a principal, the canister uses its own threshold ECDSA public key to derive deterministically a new public key for
+each requested principal. To retrieve your Ethereum address, you can call the `ethereum_address` method on the
+previously deployed canister:
+
+```shell
+dfx canister --network ic call basic_ethereum ethereum_address
```

Comment:
An alternative would be to call the function in the Candid Web UI, right?

Reply:
From the Candid Web UI you would need to set explicitly the principal, since calling it for the anonymous principal is disallowed.